src/Confluent.Kafka/Timestamp.cs

```diff
@@ -85,12 +87,13 @@ public override int GetHashCode()
         /// </param>
         /// <returns>
         ///     The milliseconds unix timestamp corresponding to <paramref name="dateTime"/>
+        ///     rounded to the previous millisecond if dateTime precision is below millisecond
         /// </returns>
         public static long DateTimeToUnixTimestampMs(DateTime dateTime)
         {
             checked
             {
-                return (long)(dateTime.ToUniversalTime() - UnixEpochUtcDateTime).TotalMilliseconds;
+                return dateTime.ToUniversalTime().Ticks / TimeSpan.TicksPerMillisecond - UnixEpochMilliseconds;
```

Comment:
> I'm wondering if we should keep the DateTime constructor or simply remove it (depreciate first). It should never be used and is misleading with the datetime comparaison

can you argue further why it should never be used and why it's misleading :-) - i can't see this atm.